storage/rocksdb/ha_rocksdb.cc

```diff
@@ -1539,8 +1539,7 @@ class Rdb_transaction
                                                 false /* just statement */);
       return HA_ERR_LOCK_DEADLOCK;
     }
-
-    if (s.IsBusy())
+    else if (s.IsBusy())
```

Comment:
IsBusy() includes IsDeadlock(). On real deadlock (IsDeadlock == true), it's not a snapshot conflict error. So "(s.Isbusy() && !s.IsDeadlock())" or "else if (s.IsBusy())" is more accurate.